### PR TITLE
add plugin foreman_host_extra_validator

### PR DESCRIPTION
**Pull request by timogoebel**
Added a very tiny plugin that allows you to validate hostnames via a custom regex.

* [x] Nightly
* [x] 1.12
* [x] 1.11
* [ ] 1.10

### CODE REVIEW

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+%global gem_name foreman_host_extra_validator
+
+%global foreman_dir /usr/share/foreman
+%global foreman_bundlerd_dir %{foreman_dir}/bundler.d
+
+Summary:    Foreman plugin to add extra validations to a host.
+Name:       %{?scl_prefix}rubygem-%{gem_name}
+Version:    0.0.4
+Release:    2%{?foremandist}%{?dist}
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: 1

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 2]
```diff
@@ -0,0 +1,82 @@
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name foreman_host_extra_validator
+
+%global foreman_dir /usr/share/foreman
+%global foreman_bundlerd_dir %{foreman_dir}/bundler.d
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: These two aren't needed when you have foreman-plugin as a BR, they're provided.

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Summary:    Foreman plugin to add extra validations to a host.
+Name:       %{?scl_prefix}rubygem-%{gem_name}
+Version:    0.0.4
+Release:    2%{?foremandist}%{?dist}
+Group:      Applications/System
+License:    GPLv3
+URL:        https://github.com/FILIADATAGmbH/foreman_host_extra_validator
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: Note: update the gemspec sometime to use this (or actually https://github.com/FILIADATAGmbH/foreman_host_extra_validator)

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires:   foreman >= 1.11.0
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: Note: update engine.rb to specify this if that's what the plugin needs, else I think just say `foreman`

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Requires:   %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems
+
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems
+BuildRequires: foreman-plugin >= 1.10.0
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: 1.11.0 to match the runtime requirement

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+Version:    0.0.4
+Release:    2%{?foremandist}%{?dist}
+Group:      Applications/System
+License:    GPLv3
+URL:        https://github.com/FILIADATAGmbH/foreman_host_extra_validator
+Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires:   foreman >= 1.11.0
+Requires:   %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: Nitpick, indentation's off

File: rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+%{foreman_restart}
+exit 0
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/app
+%{gem_instdir}/lib
+%exclude %{gem_cache}
+%{gem_instdir}/locale
+%{gem_instdir}/Rakefile
```
Comment (domcleal) [inline rubygem-foreman_host_extra_validator/foreman_host_extra_validator.spec]: Rakefile can be moved to %files doc, it's not needed at runtime

**Comment by timogoebel** [comment]
@domcleal : Made the updates. Thanks. Your suggestions to the plugin itself will be in the next release.

**Comment by domcleal** [comment]
Thanks @timogoebel, I'll build it now for the three releases.

There are also some gem2rpm templates in this repo that might help you in future, see the README for the instructions.

**Pull request merged**